packages/perseus-linter/src/rule.ts

```diff
@@ -127,13 +127,30 @@ import Selector from "./selector";
 
 import type {TraversalState, TreeNode} from "./tree-transformer";
 
+export type MakeRuleOptions = {
+    name: string;
+    pattern?: RegExp;
+    severity?: number;
+    selector?: string;
+    locale?: string;
+    message?: string;
+    lint?: (
+        state: TraversalState,
+        content: string,
+        nodes: any,
+        match: any,
+        context: LintRuleContextObject,
+    ) => string | undefined;
+    applies?: AppliesTester;
+};
+
 // This represents the type returned by String.match(). It is an
 // array of strings, but also has index:number and input:string properties.
 // TypeScript doesn't handle it well, so we punt and just use any.
-export type PatternMatchType = any;
+type PatternMatchType = any;
```

Comment:
It's too bad this is still stuck as any, but it's understandable when reading the comment above. :(